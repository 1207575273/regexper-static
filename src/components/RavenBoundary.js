import React from 'react';
import PropTypes from 'prop-types';

import RavenError from './RavenError';

class RavenBoundary extends React.Component {
  constructor() {
    super();
    this.state = {
      error: null
    };
  }

  componentDidCatch(error, errorInfo) {
    this.setState({ error, errorInfo });
  }

  render() {
    const { error, errorInfo } = this.state;
    const { children } = this.props;

    if (error) {
      const errorProps = {
        heading: 'An error has occurred.',
        details: { extra: errorInfo },
        error
      };

      return <RavenError { ...errorProps }/>;
    } else {
      return children;
    }
  }
}

RavenBoundary.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default RavenBoundary;
