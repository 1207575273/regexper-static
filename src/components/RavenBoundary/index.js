import React from 'react';
import PropTypes from 'prop-types';

import RavenError from 'components/RavenError';

class RavenBoundary extends React.Component {
  state = {
    error: null
  }

  componentDidCatch(error, errorInfo) {
    this.setState({ error, errorInfo });
  }

  render() {
    const { error, errorInfo } = this.state;
    const { children } = this.props;

    if (error) {
      const errorProps = {
        details: { extra: errorInfo },
        error
      };

      return <RavenError { ...errorProps }/>;
    } else {
      return children;
    }
  }
}

RavenBoundary.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default RavenBoundary;
