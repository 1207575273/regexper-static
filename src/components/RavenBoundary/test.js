import React from 'react';
import { shallow } from 'enzyme';

import RavenBoundary from 'components/RavenBoundary';

const testError = { error: 'test error' };
const testDetails = { details: 'test details' };

describe('RavenBoundary', () => {
  test('rendering', () => {
    const Child = () => <b>Child</b>;
    const component = shallow(
      <RavenBoundary>
        <Child/>
      </RavenBoundary>
    );
    expect(component).toMatchSnapshot();
  });

  test('rendering (with error)', () => {
    const Child = () => <b>Child</b>;
    const component = shallow(
      <RavenBoundary>
        <Child/>
      </RavenBoundary>
    );
    expect(component).toMatchSnapshot();
    component.instance().componentDidCatch(testError, testDetails);
    component.update();
    expect(component).toMatchSnapshot();
  });
});
