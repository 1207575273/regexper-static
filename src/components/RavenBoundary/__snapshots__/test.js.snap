// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RavenBoundary rendering (with error) 1`] = `<Child />`;

exports[`RavenBoundary rendering (with error) 2`] = `
<Translate(RavenError)
  details={
    Object {
      "extra": Object {
        "details": "test details",
      },
    }
  }
  error={
    Object {
      "error": "test error",
    }
  }
/>
`;

exports[`RavenBoundary rendering 1`] = `<Child />`;
